src/System.Net.Sockets/src/System/Net/Sockets/Socket.Windows.cs

```diff
@@ -166,18 +166,27 @@ internal static void SocketListToFileDescriptorSet(IList socketList, Span<IntPtr
             fileDescriptorSet[0] = (IntPtr)count;
             for (int current = 0; current < count; current++)
             {
-                if (!(socketList[current] is Socket))
+                if (!(socketList[current] is Socket socket))
                 {
                     throw new ArgumentException(SR.Format(SR.net_sockets_select, socketList[current].GetType().FullName, typeof(System.Net.Sockets.Socket).FullName), nameof(socketList));
                 }
 
-                fileDescriptorSet[current + 1] = ((Socket)socketList[current])._handle.DangerousGetHandle();
+                bool success = false;
+                socket.SafeHandle.DangerousAddRef(ref success);
```

Comment:
With your other PR, accessing socket.SafeHandle is now going to mark the handle as exposed.  Is that necessary?  Should we instead be using _handle here and everywhere else internally where the handle is used?

Reply:
I've just added an `InternalSafeHandle` property to the other PR. Once it is merged, I'll rebase and use it here.